libnetwork/drivers/windows/imanager/imanager.go

```diff
@@ -0,0 +1,95 @@
+package imanager
+
+import (
+	"github.com/docker/docker/libnetwork/datastore"
+	"github.com/docker/docker/libnetwork/discoverapi"
+	"github.com/docker/docker/libnetwork/driverapi"
+	"github.com/docker/docker/libnetwork/types"
+)
+
+const networkType = "internal"
+
+type driver struct{}
+
+// Init registers a new instance of internal manager driver.
+//
+// Deprecated: use [Register].
+func Init(dc driverapi.DriverCallback, config map[string]interface{}) error {
+	return Register(dc, config)
+}
```

Comment:
> And I would be super grateful if you were to replace all the existing driver Init calls in Swarmkit with Register at the same time.

@corhere With builtin drivers that would be maybe possible but I have no idea how to solve challenge that `Register` in `remote` wants parameters on totally different format:
https://github.com/moby/moby/blob/3ba527d82af53cf752aa6ec39daac14626c41b0f/libnetwork/drivers/remote/driver.go#L41
and whole initializer system uses some Go magic which I not familiar with.

> please don't continue to make the problem worse. Please do something to deduplicate the code. 

I'm sorry, I did notice that this will potentially things a bit worse in libnetwork but that kind of refactoring goes beyond of my coding skills.

Reply:
> I have no idea how to solve challenge that `Register` in `remote` wants parameters on totally different format

Inside [`func initializeDrivers`](https://github.com/moby/swarmkit/blob/b7708a5e9940216ceb78f76bfed668789100fe56/manager/allocator/cnmallocator/networkallocator.go#L981-L988), the register call would be written
```go
remote.Register(reg, reg.GetPluginGetter())
```

> and whole initializer system uses some Go magic which I not familiar with.

It's far less magical than you'd think. Much like with stage magic, it's all misdirection. Or in this case, indirection. Once you peel back all the layers of ceremony, registering a driver is just a call to `RegisterDriver`, which adds the driver to a map. The real trick is recognizing that all the ceremony of `initializers` and `Init` and `Register` functions is not actually very important and can be replaced wholesale without losing anything of value. If you're still interested in trying, I'll be happy to answer your questions and provide guidance. Feel free to reach out to me on the Docker Community Slack workspace.

> I'm sorry, I did notice that this will potentially things a bit worse in libnetwork but that kind of refactoring goes beyond of my coding skills.

I believe in you! But I totally understand if you feel out of your depth. I'll update this PR.

Reply:
@corhere now I don't see anything windows specific on that manager (except build flag limit but it should not be needed be be because there is no references to it yet) so would it possible have just one common one and refactoring swarmkit to use it with Linux too?

Reply:
The call to `IsBuiltinLocalDriver` in the `RegisterManager` function makes it Windows-specific. It's only there to guard against registering manager drivers with nonsense or typoed networkTypes. It would be possible to have just one common manager driver which all platforms can share if that guardrail was removed. These manager drivers are not used by any code in the moby/moby repo, so I would want to move them into the Swarmkit repo if we went in that direction. That would also make it significantly easier to plumb into Swarmkit as you wouldn't have to re-vendor moby/moby to pull in the manager drivers. What do you think of that plan? Close this PR, take the manager.go file from this branch over into your Swarmkit branch, remove the `IsBuiltinLocalDriver` guardrail and wire it up for all supported network types on all platforms, including the existing Linux ones. (Don't worry about removing the existing manager drivers from this repo; they can be abandoned-in-place for now and cleaned up later.)